modules/examples/pages/defi.adoc

```diff
@@ -0,0 +1,81 @@
+# DEFI Example
+To enable DEFI application on the IC, canisters need to interact with token canisters and the ledger canister. This example dApp provides an example on how to facilitate these interactions.
+
+The example exchange is implemented in https://github.com/dfinity/examples/tree/master/motoko/defi[Motoko] and https://github.com/dfinity/examples/tree/master/rust/defi[Rust]. 
+
+## Architecture
+The design of the IC allows for more complex on-chain computation. In combination with cheap storage, it is possible to have on-chain order books. This example takes advantage of these features and stores user balances and orders inside the exchange canister. The example exchange functionality can be condensed into the following steps:
+
+* Exchange takes custody of funds (different mechanism for tokens and ICP, see below).
+* Exchange updates internal balance book.
+* Users trade on exchange causing the exchange to update its internal balance book.
+* Withdrawing funds from the exchange gives custody back to the user.
+
+### Interface
+
+Request user-specific ledger account identifier from the exchange. This unique account identifier represents a user-specific subaccount in the exchange's ledger account, allowing it to differentiate between user deposits.
+```candid
+getDepositAddress: () -> (blob);
+```
+
+Initate user deposit to exchange. If the user wants to deposit ICP, the exchange moves the funds from the user-specific deposit address to its default subaddress and adjusts the user's ICP balance on the DEX. If the user wants to deposit a DIP token, the exchange tries to move the approved funds to its token account and adjusts the user's balance.
+```candid
+deposit: (Token) -> (DepositReceipt);
+```
+
+Withdraw request to the exchange. The exchange will send funds back to the user if the user has a sufficient balance.
+```candid
+withdraw: (Token, nat, principal) -> (WithdrawReceipt);
+```
+
+Place new order to exchange. If the order matches an existing order, it will get executed.
+```candid
+placeOrder: (Token, nat, Token, nat) -> (OrderPlacementReceipt);
+```
+
+Allows the user to cancel submitted orders.
+```candid
+cancelOrder: (OrderId) -> (CancelOrderReceipt);
+```
+
+Request user's balance on exchange for a specific token.
+```candid
+getBalance: (Token) -> (nat) query;
+```
+
+### Fee
+
+It is the responsibility of the exchange to subtract fees from the trades. This is important because the exchange must pay fees for withdraws and internal transfers.
```

Comment:
```suggestion
It is the responsibility of the exchange to subtract fees from the trades. This is important because the exchange must pay fees for withdrawals and internal transfers.
```